node/src/reactor/joiner.rs

```diff
@@ -446,9 +460,9 @@ pub(crate) struct Reactor {
     block_header_and_finality_signatures_by_height_fetcher: Fetcher<BlockHeaderWithMetadata>,
     /// This is set to `Some` if the node should shut down.
     exit_code: Option<ExitCode>,
-    // Handles request for fetching tries from the network.
-    #[data_size(skip)]
-    trie_fetcher: Fetcher<Trie<Key, StoredValue>>,
+    trie_or_chunk_fetcher: Fetcher<TrieOrChunkedData>,
```

Comment:
So what does `trie_or_chunk_fetcher: Fetcher` do? If `trie_fetcher` requests tries from the network.

Reply:
It fetches single chunks. `trie_fetcher` depends on it, as it uses this fetcher to fetch chunks.